catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBricksTest.java

```diff
@@ -43,98 +43,103 @@
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
 
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.Espresso.pressBack;
 import static android.support.test.espresso.action.ViewActions.clearText;
 import static android.support.test.espresso.action.ViewActions.click;
-import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
-import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.action.ViewActions.typeTextIntoFocusedView;
 import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
 
 import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
 
-@RunWith(AndroidJUnit4.class)
+@Category({Cat.AppUi.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
 public class BroadcastBricksTest {
-	private String defaultMessage = "defaultMessage";
-	private int broadcastSendPosition = 1;
-	private int broadcastReceivePosition = 2;
-
 	@Rule
 	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
 			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
 
+	private static int whenBrickPosition = 0;
+	private static int broadcastSendPosition = 1;
+	private static int broadcastReceivePosition = 2;
+	private static int broadcastAndWaitPosition = 3;
+
+	@Parameters(name = "{2}")
+	public static Iterable<Object[]> data() {
```

Comment:
please make this a true parametrized test that also only sets up and tests one thing at a time.

so, initialize only one brick at a time in setup, test this very brick.

because when we would add another broadcast brick at some point this test has to be changed in about 10 places, same if a brick is removed. thats not the point of parametrized tests in the first place...